framework/permissions.py

```diff
@@ -130,11 +175,20 @@ def check_login(self, *args, **kwargs):
   return check_login
 
 
-def require_edit_feature(handler):
-  """Handler decorator to require the user can edit the current feature."""
-  # TODO(jrobbins): make this per-feature
-  def check_login(self, *args, **kwargs):
-    return _reject_or_proceed(
-        self, handler, args, kwargs, can_edit_any_feature)
+def validate_feature_edit_permission(obj, feature_id):
```

Comment:
Yes, it might be better to retain the name `handler_obj` as it does in `_reject_or_proceed()`.

As for keeping this in decorator form, I couldn't find a consistent approach that was possible in that regard. From what I could tell, it is not possible to pass a dynamic parameter to a decorator function, and we now need to know the feature ID to check if the user has permission to view that feature (we have to check if they're an owner, editor, or creator).
I originally had an implementation that changed `_reject_or proceed()` to check `handler_kwargs` to see if the original function was passed a `feature_id` argument, and pass it into the `perm_function` invocation. This ended up feeling like a bad practice and odd corner cases kept popping up when running, and in testing. The current implementation was what I settled on for simplicity's sake.